src/main/java/org/spongepowered/api/text/TextBuilder.java

```diff
@@ -0,0 +1,1124 @@
+     * @see Text#getChildren()
+     */
+    public TextBuilder remove(Text... children) {
+        for (Text child : checkNotNull(children, "children")) {
+            this.children.remove(child);
+        }
+        return this;
+    }
+
+    /**
+     * Removes the specified {@link Text}s from this builder.
+     *
+     * @param children The texts to remove
+     * @return This text builder
+     * @see Text#getChildren()
+     */
+    public TextBuilder remove(Iterable<? extends Text> children) {
+        checkNotNull(children, "children");
+        if (children instanceof Collection) {
+            this.children.removeAll(asCollection(children));
+        } else {
+            for (Text child : children) {
+                this.children.remove(child);
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Removes all children from this builder.
+     *
+     * @return This text builder
+     * @see Text#getChildren()
+     */
+    public TextBuilder removeAll() {
+        this.children.clear();
+        return this;
+    }
+
+    /**
+     * Builds an immutable instance of the current state of this text builder.
+     *
+     * @return An immutable {@link Text} with the current properties of this
+     *         builder
+     */
+    public abstract Text build();
+
+    @Override
+    public boolean equals(@Nullable Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof TextBuilder)) {
+            return false;
+        }
+
+        TextBuilder that = (TextBuilder) o;
+        return Objects.equal(this.color, that.color)
+                && Objects.equal(this.style, that.style)
+                && Objects.equal(this.clickAction, that.clickAction)
+                && Objects.equal(this.hoverAction, that.hoverAction)
+                && Objects.equal(this.shiftClickAction, that.shiftClickAction)
+                && Objects.equal(this.children, that.children);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(this.color, this.style, this.clickAction, this.hoverAction, this.shiftClickAction, this.children);
+    }
+
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(TextBuilder.class)
+                .add("color", this.color)
+                .add("style", this.style)
+                .add("children", this.children)
+                .add("clickAction", this.clickAction)
+                .add("hoverAction", this.hoverAction)
+                .add("shiftClickAction", this.shiftClickAction)
+                .toString();
+    }
+
+    /**
+     * Represents a {@link TextBuilder} creating immutable {@link Text.Literal}
+     * instances.
+     *
+     * @see Text.Literal
+     */
+    public static class Literal extends TextBuilder {
+
+        protected String content;
+
+        /**
+         * Constructs a new empty {@link Literal}.
+         */
+        public Literal() {
```

Comment:
Again with consistency. How come this is public but the Text.Literal empty constructor isn't?